[리액트 7기 김주빈] 로그인 & 리뷰작성 과제 제출

KJoobin opened the pull request:
현생이 바쁘고, 강의시간이 길어서 많이 진행하지 못했습니다.. ㅠㅠ 
더 화이팅 하겠습니다!

hannut91 committed:
Move index.html file to public directory

KJoobin committed:
HomePage Login link 테스트

KJoobin committed:
HomePage login link 구현

KJoobin committed:
App에 login path 테스트

KJoobin committed:
LoginPage 구현

KJoobin committed:
LoginPage login form 테스트

KJoobin committed:
LoginForm 구현

KJoobin committed:
LoginFormContainer 와 LoginForm 연결

KJoobin committed:
LoginForm onChange 테스트

KJoobin committed:
LoginForm onChange 구현

KJoobin committed:
LoginPage log in button 테스트

KJoobin committed:
LoginForm log in button 구현

KJoobin committed:
LoginForm onSubmit 테스트

KJoobin committed:
LoginForm onSubmit 구현

KJoobin committed:
LoginForm 에서 InputField 로 분리 리팩토링

KJoobin committed:
사용하지 않는 function 삭제

KJoobin committed:
setLoginField reducer 테스트

KJoobin committed:
setLoginField reducer 구현

KJoobin committed:
setLoginField 테스트 보강

KJoobin committed:
LoginFormContainer setLoginField dispatch 테스트, 구현

KJoobin committed:
requestSession action 테스트

KJoobin committed:
requestSession actions 구현

KJoobin committed:
사용하지 않는 코드 삭제

KJoobin committed:
setAccessToken 테스트

KJoobin committed:
setAceessToken 구현

KJoobin committed:
reducer initialState 수정

KJoobin committed:
LoginFormContainer onClick event 테스트

KJoobin committed:
LoginFormContainer 에 log in event 구현

moonkii commented on src/InputField.jsx:
```diff
@@ -0,0 +1,21 @@
+export default function LoginForm({
+  onChange, label, name, type = 'text',
+}) {
```
`value` 속성도 필요할거에요.

moonkii commented on src/LoginForm.jsx:
```diff
@@ -0,0 +1,29 @@
+        name="email"
+        onChange={onChange}
+      />
+
+      <InputField
+        label="Password"
+        type="password"
+        name="password"
+        onChange={onChange}
+      />
+
```
불필요한 빈 줄은 삭제해주세요!

moonkii commented on src/LoginFormContainer.test.jsx:
```diff
@@ -0,0 +1,71 @@
+describe('LoginFormContainer', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    given('email', () => '');
+    given('password', () => '');
+  });
```
이 부분은 필요한가요?

KJoobin replied:
이후에 value 테스트 진행하려고 넣어 놨던거 같습니다.. 지우고, 나중에 테스트 할때 추가하겠습니다!

moonkii commented on src/LoginPage.test.jsx:
```diff
@@ -0,0 +1,36 @@
+
+  useSelector.mockImplementation((selector) => selector({
+    loginField: {
+      email: given.email,
+      password: given.password,
+    },
```
`given` 을 써주는 이유가 있나요?

KJoobin replied:
이부분은 given 이 필요없는거같습니다! 지우겠습니다!

moonkii commented on src/actions.js:
```diff
@@ -85,3 +93,20 @@ export function loadRestaurant({ restaurantId }) {
+  return async (dispatch, getState) => {
+    const { loginField: { email, password } } = getState();
+
+    const accessToken = await pushLogin({ email, password });
```
`pushLogin` 이라는 표현은 어색한 것 같아요. 더 의도가 드러나는 표현이 있을까요?

KJoobin replied:
네이밍을 고민하다가, 강의에서 처럼 `postLogin` 쓰려다가 실수한거같네요,,, 일단 `postLogin` 으로 진행하고 나더 좋은 방법이 생각나면 리팩토링 하겠습니다!

moonkii commented on src/actions.test.js:
```diff
@@ -100,4 +102,23 @@ describe('actions', () => {
+      });
+    });
+
+    it('dispatch requrestSession', async () => {
```
it 은 3인칭 단수이기 때문에 dispatches 가 문법상 맞겠죠?

moonkii commented on src/reducer.test.js:
```diff
@@ -125,4 +132,59 @@ describe('reducer', () => {
+      },
+    };
+
+    it('returns email value', () => {
```
`return` 해주는게 기대하는 결과인가요?

KJoobin replied:
`changes email value` 로 수정했습니다!!

moonkii reviewed (commented):
<img width="306" alt="image" src="https://user-images.githubusercontent.com/46776329/166490058-0f8cdb80-7daf-4cdd-9a1d-7094ccbfce3e.png">

바쁘신 와중에도 꾸준히 과제를 하시는게 대단하신 것 같아요 👍 
과제를 하실때 네이밍에서 의도가 잘 드러나는지 전체적을로 체크해보시면 도움되실거에요!

KJoobin committed:
불필요한 코드 삭제

- remove given in LoginFormContainer, Loginpage test
- remove line-break without reason

KJoobin committed:
테스트 it 설명 변경

- dispatch -> dispatchs
- return values -> changes field value

KJoobin committed:
LoginForm fields 테스트

KJoobin committed:
LoginForm display value 구현

KJoobin committed:
API naming 변경

- push login -> post login

KJoobin committed:
reducer state, action naming 변경 리팩토링

- loginField -> loginFields
- setLoginField -> setLoginFields

KJoobin committed:
LoginForm login했을경우 테스트

KJoobin committed:
login 시 log out button 렌더 구현

KJoobin committed:
로그인시, E-mail, Password input field not render 테스트

KJoobin commented:
강의를 보다 궁금한게 있어서 남깁니다! 저는 비즈니스 로직 - 뷰 로직 (Container 와 Presenter) 는 1 on 1 매칭을 해야한다고 생각을 했는데 강의에서는 1개의 Container 에 2개의 뷰 로직을 연결하시던데, 나중에 리팩토링을 염두해두고 (red - green - refactoring 패턴? ) 그렇게 작업을 하시는걸까요?  

아니면 로직에 따라서 1개의 Container 에 여러개의 뷰 로직을 연결해도 되는걸까요?

```js
// LoginFormContainer

{isLogin ? (
  <LoginForm { ... } />
) : (
  <LogoutForm {... } />
)}
```

ex) 7강 리뷰작성하기 57분30초 경...
<img width="543" alt="image" src="https://user-images.githubusercontent.com/45376611/166585982-783677c3-27ed-4c9d-a5b6-d5efe9ba10f9.png">

KJoobin committed:
Login form 에 onLogout 테스트

KJoobin committed:
onLogout 구현

KJoobin committed:
login form test 에 불필요한 context 삭제

- with input fields value context 삭제

KJoobin committed:
LoginFormContainer renders login fields 테스트

KJoobin committed:
render loginFields 구현

KJoobin committed:
LoginFormContainer 로그인시 테스트

KJoobin committed:
login시 구현

KJoobin committed:
logout button 을 클릭했을시 테스트

KJoobin committed:
log out 버튼 클릭스 clear session 테스트, 구현

KJoobin committed:
push login api method POST 로 수정

KJoobin committed:
fix api POST 메소드 body type

- JSON.stringify 로 body 를 parsing 해야한다.

KJoobin committed:
postLogin test

moonkii commented:
> 1개의 Container 에 여러개의 뷰 로직을 연결해도 되는걸까요?

1 on 1 을 해야한다는 원칙 같은건 없어요.
필요에 따라 뷰로직이 2개가 될 수도 있죠. 중요한건 어떤 상황인가와 코드가 말이 되는가  인 것  같아요.

moonkii commented:
CI 가 깨지는 부분을 확인하고 마무리까지 해주시면 될 것 같아요 👍

KJoobin committed:
default args 테스트

KJoobin committed:
log in, log out 텍스트 upper case 로 변경

- log in -> Log In
- log out -> Log out

KJoobin committed:
render reviews test in RestaurantDetail

KJoobin committed:
reder reviews 구현

KJoobin committed:
Reviews 테스트

KJoobin committed:
Reviews 구현

KJoobin committed:
review 가 없을경우 테스트, 구현

KJoobin committed:
Restaurant detail 에 reviews 분리 리팩토링

KJoobin committed:
RestaurantContainer 에 renders menuItems, reviews 테스트

KJoobin committed:
fixture restaurant 에 reviews 추가

KJoobin committed:
fix restaurant line-break use br tag

KJoobin committed:
review form label 테스트

KJoobin committed:
RestaurantDetail 에서 RestaurantReviews 분리 리팩토링

KJoobin committed:
review form 라벨 테스트

KJoobin committed:
review input fields 구현

KJoobin committed:
review form change event 테스트

KJoobin committed:
reivew form test 중복사용 render 만들기

KJoobin committed:
평가 남기기 버튼 테스트

KJoobin committed:
평가 남기기 버튼 구현

KJoobin committed:
click 평가 남기기 테스트

KJoobin committed:
on submit 평가 남기기 구현

KJoobin committed:
render field values 테스트

KJoobin committed:
with out access token 일때 dose not render review form 테스트

KJoobin committed:
access token 이 있을때만 review form render 구현

KJoobin committed:
Restaurant container review form chagne event 테스트

KJoobin committed:
setReviewFields 구현

KJoobin committed:
setReviewFields 테스트 보강, 오타 수정

KJoobin committed:
평가 남기기 called dispatch 테스트

KJoobin committed:
submit review 테스트, 구현

KJoobin committed:
add storage, mock storage

KJoobin committed:
remove console

KJoobin committed:
App when logged in test

KJoobin committed:
get access token when logged in 구현

KJoobin committed:
when logged out test

KJoobin committed:
setItems with accessToken when request session 테스트, 구현

KJoobin committed:
storage 테스트

KJoobin committed:
fix typo 평가 남기기 -> 리뷰 남기기

KJoobin committed:
clear storage when clear session test

KJoobin committed:
clear storage when clear session 구현

KJoobin committed:
mockClear naming refactoring

KJoobin committed:
saveItem -> setItem naming refactoring

moonkii commented on src/App.jsx:
```diff
@@ -4,13 +4,27 @@ import {
   Link,
 } from 'react-router-dom';
 
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import HomePage from './HomePage';
 import AboutPage from './AboutPage';
+import LoginPage from './LoginPage';
 import RestaurantsPage from './RestaurantsPage';
 import RestaurantPage from './RestaurantPage';
 import NotFoundPage from './NotFoundPage';
+import { getItem } from './services/storage';
+import { setAccessToken } from './actions';
```
빈 줄을 넣어 의미상으로 구분을 해주면 코드가독성을 높이는데 좋을 것 같아요!

moonkii commented on src/App.jsx:
```diff
@@ -4,13 +4,27 @@ import {
 export default function App() {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const accessToken = getItem('accessToken');
+
+    if (accessToken) {
+      dispatch(setAccessToken({ accessToken }));
+    }
+  });
```
공식문서를 참고하시면 2번째 인자가 필요하다는걸 알 수 있을거에요.

moonkii commented on src/LoginFormContainer.test.jsx:
```diff
@@ -0,0 +1,107 @@
+
+describe('LoginFormContainer', () => {
+  beforeEach(() => {
+    given('loginFields', () => ({
+      email: '',
+      password: '',
+    }));
+
+    given('accessToken', () => '');
```
이 부분들이 필요한가요?

KJoobin replied:
옙! `context('with login'), context('without login')` 체크랑, `renders login fields ` 를 테스트 하기 위해서 필요합니다!

moonkii commented on src/LoginPage.jsx:
```diff
@@ -0,0 +1,11 @@
+  return (
+    <div>
+      <h2>Login 페이지</h2>
+
```
불필요한 빈 줄은 삭제해주세요.
셀프리뷰를 하시면서 이런 부분들을 찾아주시면 좋아요.

KJoobin committed:
불필요한 빈줄 삭제, 필요한 빈줄 추가

KJoobin committed:
useEffect dependency deps 추가

- App.jsx 에 accessToken 을 불러오는 useEffect 에 dependency deps 추가

KJoobin committed:
의미상 구분을 위해 빈줄 추가
